python_modules/libraries/dagster-k8s/dagster_k8s/launcher.py

```diff
@@ -175,72 +180,20 @@ def from_config_value(cls, inst_data, config_value):
         return self._inst_data
 
-    def get_static_job_config(self):
-        if self._job_config:
-            return self._job_config
-        else:
-            self._job_config = DagsterK8sJobConfig(
-                job_image=check.str_param(self._job_image, "job_image"),
-                dagster_home=check.str_param(self.dagster_home, "dagster_home"),
-                image_pull_policy=check.str_param(self._image_pull_policy, "image_pull_policy"),
-                image_pull_secrets=check.opt_list_param(
-                    self._image_pull_secrets, "image_pull_secrets", of_type=dict
-                ),
-                service_account_name=check.str_param(
-                    self._service_account_name, "service_account_name"
-                ),
-                instance_config_map=check.str_param(
-                    self.instance_config_map, "instance_config_map"
-                ),
-                postgres_password_secret=check.opt_str_param(
-                    self.postgres_password_secret, "postgres_password_secret"
-                ),
-                env_config_maps=check.opt_list_param(
-                    self._env_config_maps, "env_config_maps", of_type=str
-                ),
-                env_secrets=check.opt_list_param(self._env_secrets, "env_secrets", of_type=str),
-                env_vars=check.opt_list_param(self._env_vars, "env_vars", of_type=str),
-                volume_mounts=self._volume_mounts,
-                volumes=self._volumes,
-                labels=self._labels,
-            )
-            return self._job_config
-
-    def _get_grpc_job_config(self, job_image):
-        return DagsterK8sJobConfig(
-            job_image=check.str_param(job_image, "job_image"),
-            dagster_home=check.str_param(self.dagster_home, "dagster_home"),
-            image_pull_policy=check.str_param(self._image_pull_policy, "image_pull_policy"),
-            image_pull_secrets=check.opt_list_param(
-                self._image_pull_secrets, "image_pull_secrets", of_type=dict
-            ),
-            service_account_name=check.str_param(
-                self._service_account_name, "service_account_name"
-            ),
-            instance_config_map=check.str_param(self.instance_config_map, "instance_config_map"),
-            postgres_password_secret=check.opt_str_param(
-                self.postgres_password_secret, "postgres_password_secret"
-            ),
-            env_config_maps=check.opt_list_param(
-                self._env_config_maps, "env_config_maps", of_type=str
-            ),
-            env_secrets=check.opt_list_param(self._env_secrets, "env_secrets", of_type=str),
-            env_vars=check.opt_list_param(self._env_vars, "env_vars", of_type=str),
-            volume_mounts=self._volume_mounts,
-            volumes=self._volumes,
-            labels=self._labels,
-        )
+    def get_container_context(self, pipeline_run: PipelineRun) -> K8sContainerContext:
```

Comment:
More explicit? since a run launcher could have its own context, without a run
```suggestion
    def get_container_context_for_run(self, pipeline_run: PipelineRun) -> K8sContainerContext:
```